sys/net/include/fib/fib.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2014  Martin Landsmann <Martin.Landsmann@HAW-Hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_fib Forwarding Information Base (FIB)
+ * @brief       FIB implementation
+ * @ingroup     net
+ *
+ * @{
+ *
+ * @file        fib.h
+ * @brief       Types and functions for FIB
+ * @author      Martin Landsmann
+ */
+
+#ifndef _FIB_H
+#define _FIB_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Reactive Routing Protocol (RRP) message content to request/reply route discovery
+ */
+typedef struct rrp_address_msg_t {
+    uint8_t *address;      /**< The pointer to the address */
+    uint8_t address_size;  /**< The address size */
+    kernel_pid_t iface_id; /**< The interface used for the next-hop reply.
+                                If it is KERNEL_PID_UNDEF, no next-hop has been found. */
+    uint32_t lifetime;     /**< The lifetime of this next-hop */
+} rrp_address_msg_t;
+
+#define FIB_MSG_RRP_SIGNAL (0x99)              /**< message type for RRP notifications         */
```

Comment:
Just to sum up possible `fib_get_next_hop()` results (noted in `(iface, next_hop_addr)` notation) and there meaning:
 * `(iface, address)`: get link-layer address for `address` and send it over `iface`
 * `(iface, NULL)`: do not get link-layer address for `address` (and subsequently build no link-layer header) and send it over `iface`
 * `(KERNEL_PID_UNDEF, NULL)`: drop packet, since no route was or will be found.